File: production/shared/src/main/scala/kpn/shared/data/Node.scala
package kpn.shared.data

import kpn.shared.LatLon
import kpn.shared.data.raw.RawNode
import kpn.shared.Timestamp

case class Node(raw: RawNode) extends Element with LatLon {

  def id: Long = raw.id

  def latitude: String = raw.latitude

  def longitude: String = raw.longitude

  def version: Int = raw.version

  def timestamp: Timestamp = raw.timestamp

  def changeSetId: Long = raw.changeSetId

  def tags: Tags = raw.tags

  override def isNode: Boolean = true

}

File: backend/api/src/main/java/kpn/api/common/data/Node.java
package kpn.api.common.data;

import kpn.api.custom.Tag;
import kpn.api.custom.Timestamp;

import com.google.common.collect.ImmutableList;

public record Node(
  Long id,
  String latitude,
  String longitude,
  Long version,
  Timestamp timestamp,
  Long changeSetId,
  ImmutableList<Tag> tags
) {
}

/*
package kpn.api.common.data

import kpn.api.common.LatLon
import kpn.api.common.data.raw.RawNode
import kpn.api.custom.Tag
import kpn.api.custom.Timestamp

case class Node(
  id: Long,
  latitude: String,
  longitude: String,
  version: Long,
  timestamp: Timestamp,
  changeSetId: Long,
  tags: Seq[Tag]
) extends Element with LatLon {
  override def isNode: Boolean = true

  def toRaw: RawNode = {
    RawNode(
      id,
      latitude,
      longitude,
      version,
      timestamp,
      changeSetId,
      tags
    )
  }
}

*/
